plugin/completion.py

```diff
@@ -181,22 +282,23 @@ def format_completion(self, item: dict, can_resolve_completion_items: bool) -> s
 
         lsp_label = item["label"]
         lsp_filter_text = item.get("filterText")
-        lsp_detail = item.get("detail") or ""
-        lsp_detail = html.escape(lsp_detail.replace('\n', ' '))
-
-        if can_resolve_completion_items or "documentation" in item:
-            doc_link = '<a href="subl:lsp_resolve_docs">Documentation</a>'
-        else:
-            doc_link = ''
+        lsp_detail = html.escape(item.get("detail", "").replace('\n', ' '))
 
         if lsp_filter_text and lsp_filter_text != lsp_label:
             st_trigger = lsp_filter_text
             st_annotation = lsp_label
-            st_details = '{} {}'.format(doc_link, lsp_detail) if doc_link else lsp_detail
         else:
             st_trigger = lsp_label
-            st_annotation = lsp_detail
-            st_details = doc_link
+            st_annotation = ""
+
+        st_details = ""
+        if can_resolve_completion_items or "documentation" in item:
+            args = {
+                "item": deep_escape(item)
+            }
+            st_details += "<a href='subl:lsp_resolve_docs {}'>Δ</a>".format(sublime.encode_value(args))
```

Comment:
How about this: Store the `response_items` from `def handle_response` in the listener (`self.response_items = ...`)

Then, encode a simple integer here as argument. I think you can use a simple string-format then:

```
'{{"i":{}}}'.format(index)
```
Then, in LspResolveDocsCommand, you can fish up the listener with 

```
for listener in sublime_plugin.view_event_listeners[self.view.id()]:
    if isinstance(listener, CompletionHandler):
        item = listener.response_items(index)
        # ... do stuff with item
```
This way we avoid calling `html.escape` for each completion item that has docs.

Reply:
Should I wait for this sublime issue to be resolved or should I implement your suggested solution.
https://github.com/sublimehq/sublime_text/issues/3318

If 3318 gets fixed, that means that I don't have to deep escape the item.

Do you still want me to implement the index approach?

Reply:
I will go with the index approach, and see how it goes

Reply:
we can easily add the new api when sublime adds it